src/requests/BaseApi.ts

```diff
@@ -0,0 +1,168 @@
+        
+        // url checking for keywords to change for query parameters
+
+        if (this.currentRequest.requiredQueryParams) {
+            // we make sure all required param are set in the query params variable
+            for (const param of this.currentRequest.requiredQueryParams) {
+                if (!this.currentQueryParams.find((element) => element.name === param)) {
+                    throw new Error("Missing required query parameter: " + param);
+                }
+            }
+        }
+        // if all query parameters are present in the query params variable then we can replace them with our parameter from the template
+        let currentPath = this.baseClass.basePath + this.currentRequest.path;
+
+        if (this.currentQueryParams.length > 0) {
+            // check for all required parameters from this.currentRequest.requestQueryParams
+            for (const param of this.currentQueryParams) {
+                currentPath = currentPath.replace(`{${param.name}}`, param.value);
+            }
+            // console.log(currentPath);
+            
+            if (currentPath.includes("{") || currentPath.includes("}")) {
+                throw new Error("Missing required query parameter!");
+            }
+        }
+
+        // now we add any path parameters
+        if (pathKeys) {
+            currentPath += "?" + pathKeys;
+        }
+
+        const url: URL = new URL(currentPath);
+
+        // check headers
+        const headers: headerOptions = {
+            "User-Agent": this.baseClass.headerAgent // we set the User_Agent header
+        }
+        // headers["User-Agent"] = this.baseClass.headerAgent; 
+
+        if (!this.currentRequest.cookiesNeeded.includes("none")) {
+            headers["Cookie"] = ``;
+
+            if (this.currentRequest.cookiesNeeded.includes("authorization")) {
+                headers['Authorization'] = `Basic ${this.baseClass.getBase64Credentials()}`;
+            }
+            if (this.currentRequest.cookiesNeeded.includes("authCookie")) {
+                headers["Cookie"] += `${this.baseClass.instanceCookie.getAuthCookie()} `;
+            }
+            if (this.currentRequest.cookiesNeeded.includes("twoFactorAuth")) {
+                headers["Cookie"] += `${this.baseClass.instanceCookie.getTwoFactorAuthCookie()} `;
+            }
+        }
+        
+        // check options
+        const options: VRCRequest = {
+            method: this.currentRequest.method,
+            headers: headers
+        };
+        if (body && this.currentRequest.requiresData) {
+            options.body = JSON.stringify(body);
+        }
+
+        // call the fetch
+        // console.log("-- BEFORE CALL --");
+        // console.log("URL: ", url);
+        // console.log("options: ", options);
+        // console.log("-- END BEFORE CALL --");
+        
+
+        try {
+            const response: API<E | RequestSuccess, RequestError> = await fetch(url, { ...options });
```

Comment:
This should be RequestSucess & E or only E, that's the point of using the type. You're doing "as E" on line 151, which removes all the purpose of the typing here.